### PR TITLE
Particular status bookings

### LINKED ISSUE #38: Get Renter bookings that have a particular status
/api/v1/renters/renter_id/bookings?status=STATUS

### PR DESCRIPTION
**Pull request by abreaux26**
Need to merge [this pr FIRST](https://github.com/Yardsourcing/yardsourcing-engine/pull/58)

added renters/:renter_id/bookings route
index method for Bookings controller inside renters folder

Testing
multiple sad path and happy paths
100%

closes issue [#38 ](https://github.com/Yardsourcing/yardsourcing-engine/issues/38)

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -5,15 +5,19 @@
       resources :purposes, only: [:index]
 
       get '/yards/yard_search', to: 'yards/search#index'
-      get    '/yards/:id/bookings', to: 'yards/bookings#index'
+      get '/yards/:id/bookings', to: 'yards/bookings#index'
       resources :yards, except: [:index] do
         # resources :bookings, only: [:index]
         end
-      resources:bookings, only: [:show]
+      resources :bookings, only: [:show]
```
Comment (jbranham1) [inline config/routes.rb:12]: Thank you for the space lol

**Review by jbranham1 — APPROVED** [review]
Great job!

File: app/models/booking.rb [hunk 2]
```diff
@@ -17,4 +17,8 @@ class Booking < ApplicationRecord
   def date_cant_be_in_the_past
     errors.add(:date, "Booking Date can't be in the past") if date && date < Date.today
   end
+
+  def self.find_by_renter_and_status(renter_id, status)
+    where('renter_id = ? and status = ?', renter_id, Booking.statuses[status.to_sym])
```
Comment (Gvieve) [inline app/models/booking.rb:22]: I haven't seen this syntax before. Very cool!

**Review by Gvieve — APPROVED** [review]
Y'all are just speeding along. Great job!

**Pull request merged**